CoreScriptsRoot/Modules/Settings/Pages/GameSettings.lua

```diff
@@ -103,10 +103,38 @@ local function Initialize()
 
 		settingsDisabledInVR[this.FullscreenEnabler] = true
 
-		local fullScreenSelectionFrame = this.FullscreenEnabler.SliderFrame and this.FullscreenEnabler.SliderFrame or this.FullscreenEnabler.SelectorFrame
-
+		local Debounce = false
+		
 		this.FullscreenEnabler.IndexChanged:connect(function(newIndex)
-			GuiService:ToggleFullscreen()
+			if Debounce then return end
+			Debounce = true
+			if newIndex == 1 then
+				if not GameSettings:InFullScreen() then
+					GuiService:ToggleFullscreen()
```

Comment:
If I take it out of the if/else then the same thing would happen as below. If I left the whole function the same as it was.